src/DisplayBoard.cpp

```diff
@@ -30,14 +38,20 @@ std::string display(const Board& b)
     auto height = b.getSizeCols();
     auto widthRows = b.getLongestCluesLenghtInRows();
     auto heightCol = b.getLongestCluesLenghtInCols();
+    const Lines cluesCols = b.getCluesCols();
+    std::vector<std::string> stringClues = getFormattedColumns(heightCol, cluesCols);
+    const Lines cluesRows = b.getCluesRows();
     std::stringstream output;
 
-    output << drawColumns(widthRows, width, heightCol);
+    output << drawColumns(widthRows, width, heightCol, stringClues);
     output << drawEndLine(widthRows, width) << "\n";
 
     for (int i = 0; i < height; i++)
     {
-        output << drawRow(widthRows);
+        if(cluesRows.size() != height)
```

Comment:
this if should be before for loop for a better performance. Now this condition is checked at every loop iteration.